lib/core.js

```diff
@@ -407,7 +407,7 @@ declare class Promise<R> {
 
     then<U>(
       onFulfill?: (value: R) => Promise<U> | U,
-      onReject?: (error: any) => Promise<U> | U
+      onReject?: (error: any) => ?Promise<U> | U
```

Comment:
Can you provide a reference for this change? I'm not super familiar with all the intricacies of the Promise API, but I'm a little suspicious of this change.